crates/tbdex/src/http_client/exchanges.rs

```diff
@@ -151,10 +151,30 @@ pub fn get_exchange(
     Ok(exchange)
 }
 
-pub fn get_exchange_ids(pfi_did: &str, requestor_did: &BearerDid) -> Result<Vec<String>> {
+#[derive(Clone, Default, Debug, PartialEq)]
+pub struct GetExchangeIdsQueryParams {
+    pub pagination_offset: Option<i64>,
+    pub pagination_limit: Option<i64>,
+}
+
+pub fn get_exchange_ids(
+    pfi_did: &str,
+    requestor_did: &BearerDid,
+    query_params: Option<GetExchangeIdsQueryParams>,
+) -> Result<Vec<String>> {
     let service_endpoint = get_service_endpoint(pfi_did)?;
     let get_exchanges_endpoint = format!("{}/exchanges", service_endpoint);
 
+    let get_exchanges_endpoint = if let Some(params) = query_params {
+        add_pagination(
```

Comment:
we can improve this in due time, but for now this is the most efficient way. namely, we may consider using a dedicated `Url` type, but which will require additional considerations (namely it may require adding a new dependency which must pass licensing considerations)

Reply:
btw we need to add pagination to other functions too, so we can improve DRY and the like at such time